### PR TITLE
OSX is alias for MacOS

### PR DESCRIPTION
**Pull request by buyaa-n**
Related to https://github.com/dotnet/runtime/issues/49323

While testing the Known Platform names analyzer in runtime repo we discovered that runtime repo is using OSX in TFM which is  alias (old name) of MacOS. As OSX can be used instead if MacOS we should add it as known platform in case MacOS exist in the known platforms list

### CODE REVIEW

**Review by jeffhandley — APPROVED** [review]

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformString.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@ public sealed class UseValidPlatformString : DiagnosticAnalyzer
         private static readonly ImmutableArray<string> methodNames = ImmutableArray.Create("IsOSPlatform", "IsOSPlatformVersionAtLeast");
         private const string IsPrefix = "Is";
         private const string VersionSuffix = "VersionAtLeast";
+        private const string macOS = nameof(macOS);
```
Comment (carlossanlop) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformString.cs:24]: Is there a place somewhere in the roslyn-analyzers repo where we have all the platform names predefined? Seems strange to have only this one defined here.

Also, wouldn't it be better to use a `const string` instead of a `nameof` that points to its own name? This pattern is a bit strange.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs [hunk 2]
```diff
@@ -376,6 +376,31 @@ public class Test
...
+
+    [SupportedOSPlatform(""MacOS1.2"")]
+    public void SupportedMacOs2PartValid() { }
+
+    [{|#1:SupportedOSPlatform(""OSX1.2.3.4"")|}] // Version '1.2.3.4' is not valid for platform 'OSX'. Use a version with 2-3 parts for this platform.
+    public void SupportedOSPlatformOSX4PartsInvalid() { }
+
+    [SupportedOSPlatform(""Osx1.2"")]
+    public void SupportedOsx2PartValid() { }
+}";
```
Comment (carlossanlop) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs:398]: What would happen if you add a method topped with two `[SupportedOSPlatform]` attributes, one with "OSX" and another one with "MacOS"? Maybe we should have a test for that edge case too.

**Review by carlossanlop — APPROVED** [review]
LGTM. Left a couple of non-blocking comments. Consider adding the extra unit test I suggested.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformString.cs (hunk 1 above)
Comment (buyaa-n) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformString.cs:24]: > Is there a place somewhere in the roslyn-analyzers repo where we have all the platform names predefined? Seems strange to have only this one defined here.

Do not know, anyway for the analyzer the known names loaded from MSBuild known platforms list

> Also, wouldn't it be better to use a const string instead of a nameof that points to its own name? This pattern is a bit strange.

It can, but i wouldn't say that would be better. In this case, as you can see the variable name is the same as the string literal (no need to be different or no need to be more descriptive), in fact, it is not just a random string literal, it illustrates that it is a known OS platform name from MSBuild list used with the exact same value and casing from the list.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs (hunk 2 above)
Comment (buyaa-n) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs:398]: Nothing should happen, the analyzer should not distinguish them as same nor should require to remove one of them.

I can add a test to show nothing happening
